scripts/buildNomisIncidentTypePayload.ts

```diff
@@ -0,0 +1,158 @@
+    printHelpAndExit(scriptName)
+    return
+  }
+
+  const dpsConfig = await getIncidentTypeConfiguration(dpsTypeCode)
+  const payload = toPrisonApiUpdatePayload({ dpsConfig, description: typeMeta.description, active: typeMeta.active })
+
+  // Helpful info for the caller
+  printText(`Endpoint: PUT {HMPPS_PRISON_API_URL}/incidents/configuration/${typeMeta.nomisCode}`)
+  printText(JSON.stringify(payload, null, 2))
+}
+
+function toPrisonApiUpdatePayload({
+  dpsConfig,
+  description,
+  active,
+}: {
+  dpsConfig: DpsIncidentTypeConfiguration
+  description: string
+  active: boolean
+}) {
+  const orderedQuestions = orderQuestionsByFlow(dpsConfig)
+  const questionsArray = orderedQuestions.map(q => ({
+    code: q.code,
+    active: q.active,
+    question: q.question,
+    multipleAnswers: q.multipleAnswers,
+    answers: q.answers.slice().map(ans => ({
+      code: ans.code,
+      response: ans.response,
+      active: ans.active,
+      commentRequired: ans.commentRequired,
+      dateRequired: ans.dateRequired,
+      nextQuestionCode: ans.nextQuestionCode,
+    })),
+  }))
+
+  return {
+    incidentTypeDescription: description,
+    active,
+    questions: questionsArray,
+    prisonerRoles: dpsConfig.prisonerRoles.map(pr => ({
+      prisonerRole: pr.prisonerRole,
+      onlyOneAllowed: pr.onlyOneAllowed,
+      active: pr.active,
+    })),
+  }
+}
+
+/**
+ * Orders questions to match the order a user would typically be presented with:
+ * - Start at the configured startingQuestionCode
+ * - Follow nextQuestionCode links from answers (preserving the order of answers)
+ * - Avoid revisiting questions (handles cycles)
+ * - Append any remaining questions not reachable from the start in a stable order
+ */
+function orderQuestionsByFlow(dpsConfig: DpsIncidentTypeConfiguration): DpsQuestionConfiguration[] {
+  const questionsByCode = dpsConfig.questions
+  const visited = new Set<string>()
+  const orderedCodes: string[] = []
+
+  const dfs = (code: string | null | undefined) => {
+    if (!code) return
+    const q = questionsByCode[code]
+    if (!q) return
+    if (visited.has(code)) return
+    visited.add(code)
+    orderedCodes.push(code)
+
+    // Determine the next questions from the current question's ACTIVE answers, preserving answer order
+    const seenNext = new Set<string>()
+    q.answers.forEach(ans => {
+      if (!ans.active) return
+      const next = ans.nextQuestionCode
+      if (!next) return
+      if (!questionsByCode[next]) return
+      if (seenNext.has(next)) return
+      seenNext.add(next)
+      dfs(next)
+    })
+  }
+
+  // Traverse from the configured starting question
+  dfs(dpsConfig.startingQuestionCode)
+
+  // Append any remaining questions (unreachable from start) in a stable order
+  const remaining = Object.keys(questionsByCode).filter(code => !visited.has(code))
+  const compareCodes = (a: string, b: string) => {
+    const ai = parseInt(a, 10)
+    const bi = parseInt(b, 10)
```

Comment:
Watch out because this is assuming the codes are numbers, it is usually the case historically but the idea is that we may want to use more human friendly identifiers long term potentially.

Although, in maybe in practice it doesn't matter for this script?